src/shared/components/NotFound.tsx

```diff
@@ -136,10 +140,23 @@ const NotFound: FC = () => {
 
   const handleDeepLink = useCallback(async () => {
     if (!org.current) {
-      setIsFetchingOrg(true)
-      org.current = await fetchOrg()
-    }
+      if (shouldUseQuartzIdentity()) {
+        try {
+          setIsFetchingOrg(true)
+          const defaultAccount = await getDefaultAccount()
+          const quartzOrgs = await fetchOrgsByAccountID(defaultAccount.id)
 
+          const deafultQuartzOrg =
+            quartzOrgs.find(org => org.isDefault) || quartzOrgs[0]
```

Comment:
Very nice work and very clean code! 🎉 

It might make for cleaner code at this level to combine these two calls into a single function inside the `identity/api/auth` layer, but I'll leave it to your judgement.